src/python/pants/engine/native.py

```diff
@@ -400,7 +400,7 @@ def extern_store_set(self, context_handle, vals_ptr, vals_len):
         """Given storage and an array of Handles, return a new Handle to represent the set."""
         c = self._ffi.from_handle(context_handle)
         return c.to_value(
-            OrderedSet(c.from_value(val[0]) for val in self._ffi.unpack(vals_ptr, vals_len))
+            FrozenOrderedSet(c.from_value(val[0]) for val in self._ffi.unpack(vals_ptr, vals_len))
```

Comment:
I'm not sure how this works. Is `extern_store_set` code to convert a Rust set into a Python object? So now we convert it into `FrozenOrderedSet`? Or is this to convert Python sets into Rust code?

cc @gshuflin

Reply:
I think this is fine. `externs::store_set` in Rust basically marshals a bunch of Rust values into a C-style array of Python `Handle`s and then invokes this `extern_store_set` function with `vals_ptr` and `vals_len` set correctly. `extern_store_set` then uses the `_ffi` library tools to unmarshal that array of `Handle`s into Python objects, and calls the object constructor with them in Python. The Rust FFI code shouldn't care whether the Python object constructor this function calls is an `OrderedSet` or a `FrozenOrderedSet`. As far as Rust is concerned the output of this function is an opaque Python `Value` no matter what.

Reply:
Actually now that I'm looking more closely at this code, I'm not sure if this is ever actually being called to begin with... The Rust extern function is annotated with `dead_code`. We don't seem to be currently ever creating a Python set from Rust.

Reply:
This is mostly annotated with `dead_code` because we didn't want to have to figure out how to do this again if we need a set later.

Reply:
Thank you both for the explanations! TIL.